featurebyte/session/databricks.py

```diff
@@ -50,15 +71,22 @@ def _initialize_connection(self) -> None:
         if not HAS_DATABRICKS_SQL_CONNECTOR:
             raise RuntimeError("databricks-sql-connector is not available")
 
-        self._connection = databricks_sql.connect(
-            server_hostname=self.host,
-            http_path=self.http_path,
-            access_token=self.database_credential.access_token,
-            catalog=self.catalog_name,
-            schema=self.schema_name,
-            _user_agent_entry=APPLICATION_NAME,
-            _use_arrow_native_complex_types=False,
-        )
+        with patch("databricks.sql.client.ThriftBackend", ThriftBackend):
```

Comment:
This patch is necessary for the query execution to be cancellable while the cursor is polling for results

Reply:
Would be good to include this as a comment in the code.